src/Security/EditableRolesBuilder.php

```diff
@@ -45,11 +46,17 @@ class EditableRolesBuilder
      */
     protected $rolesHierarchy;
 
-    public function __construct(TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authorizationChecker, Pool $pool, array $rolesHierarchy = [])
+    /**
+     * @var SonataConfiguration
+     */
+    private $configuration;
+
+    public function __construct(TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authorizationChecker, Pool $pool, SonataConfiguration $configuration, array $rolesHierarchy = [])
```

Comment:
Might be changed on 4.x branch to help the migration

Reply:
Yep, A help providing this on lower branch is very welcome. Since my time is very limited and I dont want to grow old doing the migration I would like to provide small Prs (as small as the bundle let me) with all the bc break that needs to be included.

I will focus on having high phpstan and psalm to ensure I dont break a lot of things.

Reply:
Also , keep in mind the idea is to remove FOSUser, the migration wont be easy anyway

Reply:
Indeed, might not be worth it since we remove FOSUser